ast_tools/passes/ssa.py

```diff
@@ -258,6 +263,7 @@ def leave_Return(self,
                     )
                 )
             self.added_names.add(attr_name.value)
+            self.symbol_table_skip_targets.add(attr_name)
```

Comment:
Do `added_names` and `symbol_table_skip_targets` hold effectively equivalent data? Can we just build one and use it to build the other?

Also be aware that this will hash on identity not value `Name(value='x') != Name(value='x')`

Reply:
If you want structural hashing use DeepNode

Reply:
Not quite, because we skip the added names for final, but we don't want to skip the return value assign which is included in added_names (since these correspond to the original return statement).  We can separate added_names into two sets and merge them for the other code (to avoid storing them in two sets)